enhancements/etcd/disaster-recovery-with-ceo.md

```diff
@@ -19,86 +19,200 @@ see-also:
-This script helps recover from a complete loss of a master host. This includes situations where a majority of master hosts have been lost, leading to etcd quorum loss and the cluster going offline. This procedure assumes that you have at least one healthy master host. It restores etcd data from backup to gain etcd quorum on the active master host. The steps to recover include setting up an etcd-signer to obtain certificates for new master hosts and growing them to full membership.
-
-### Recover from expired certificates ([recover-kubeconfig.sh](https://github.com/openshift/machine-config-operator/blob/3e10747951dd1c3c7a9f131d4c5af7805f19a164/templates/master/00-master/_base/files/usr-local-bin-openshift-kubeconfig-gen.yaml))
-This procedure enables to recover from a situation where the control plane certificates have expired.
-
-
 ## Proposal
+The proposal is to transfer the ownership of the DR scripts from MCO, remove all the deprecated DR scripts, and to 
+simplify other existing scripts for backing up and restoring the cluster state. 
 
-## User Stories [optional]
+With the simplification of the scripts, different disaster recovery scenarios are documented properly to utilize the
+simplified scripts along with other Openshift utility commands to achieve the recovery of the cluster. 
 
 ## Implementation Plan
-
-1. Make changes to the scripts as needed
-2. Modify CEO to provide the information needed for the DR scripts to work.
+###  Transfer the ownership of the DR files from the MCO to CEO
+All etcd-scripts should be removed from MCO, and be generated along with the static-pod-resources as configmaps. The 
+scripts should be copied to /usr/local/bin in the init container for a convenient access to the admin user.
+    
+###  Remove the deprecated DR scripts
+The scripts for adding a new etcd member; removing an etcd member; and recovering an etcd-member are no longer needed,
+and should be deleted from the repository.
+    
+### Backing up the cluster state
+The script for backing up the cluster state should back up static-pod-resources for etcd, kube-apiserver, 
+kubecontrollermanager and kube-scheduler, along with a copy of the etcd snapshot database. The reason for backing up
+the static pod resources is to match the static pods state on disk with what is in etcd,  when we restore. Furthermore,
+when etcd data is encrypted, the encryption keys are stored in static pod resources, and restoring the etcd data without
+restoring the corresponding key will make the etcd data unusable.
+    
+### Restoring the cluster state from backup 
+This solution handles situations where you want to restore your cluster to a previous state, for example, if something
+critical got deleted accidentally. This process is also used for recovering from a loss of quorum.
+    
+The script for restoring to a previous cluster state using a backup file will restore the etcd database from the backup
```

Comment:
You need to describe how this works.  Not just that you plan to do it.    Lay it out in 10ish easy steps.

Reply:
The outline of the scripts is given below in a separate section.